receiver/githubreceiver/config.go

```diff
@@ -0,0 +1,82 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubreceiver // import "github.com/liatrio/liatrio-otel-collector/receiver/githubreceiver"
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/liatrio/liatrio-otel-collector/receiver/githubreceiver/internal"
+	"github.com/liatrio/liatrio-otel-collector/receiver/githubreceiver/internal/metadata"
+)
+
+const (
+	scrapersKey = "scrapers"
+)
+
+// Config that is exposed to this github receiver through the OTEL config.yaml
+type Config struct {
+	scraperhelper.ControllerConfig `mapstructure:",squash"`
+	Scrapers                       map[string]internal.Config `mapstructure:"scrapers"`
+	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
+}
+
+var _ component.Config = (*Config)(nil)
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+// Validate the configuration passed through the OTEL config.yaml
+func (cfg *Config) Validate() error {
+	if len(cfg.Scrapers) == 0 {
+		return errors.New("must specify at least one scraper")
```

Comment:
I'm not sure if we adjusted this message, but probably should. Could be done on the next iteration.